### PR TITLE
Refactor:  Pytest refactoring

### PR DESCRIPTION
**Pull request by KonikaChaurasiya-GSLab**
## Change Summary

Pytest refactoring

## Related Issue(s)

Fixes #N/A

## Component(s) name

Pytest

## Proposed changes

Define a clear structure to separate:

- Unit test with no CV interaction to test component: `tests/unit`
- System test to run module backend component and which requires CV connection: `tests/system`
- lib to provides a safe place for fixtures, parametrizes, mook data: `tests/lib`

## How to test

```bash
# Go to test folder
$ cd tests/

# Run Pytest
$ pytest --html=report.html --self-contained-html --cov=. --cov-report=html --capture=tee-sys --color yes unit
```

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly. (check the box if not applicable)
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by guillaumeVilar** [comment]
Hello, please find my suggestions below. 

1. Modify the tests/README with the info from the PR:
* Unit test with no CV interaction to test component: tests/unit
* System test to run module backend component and which requires CV connection: tests/system
* lib to provides a safe place for fixtures, parametrizes, mook data: tests/lib

2. Remove unused import from system/*py ?

3. Add instruction on how to set a test CVP server? (the file to be modified is `system/constatns_data.py`)

4. In `ansible-cvp/tests/Makefile`: The make config.py echo message doesn't match the correct path to the config.py file. 

5. Empty file  ansible-cvp/tests/conftest.py (might be due to my environment - not sure). 

6. Unit test OK

7. System tests:  


7.1. `test_cv_container_tools.py`: function test_get_configlet_from_container is not using the variable UNIT_TEST is not used inside the function (using hard coded names).  

7.2 `test_cv_device_tools_fqdn.py`:  function test_device_is_present_by_serial (line 140) the search should be made using the FQDN: 
```
            assert self.inventory.is_device_exist(
                device_lookup=CV_DEVICE[FIELD_FQDN], search_mode=FIELD_FQDN) is True
```
Otherwise, the test will fail. 
We should also maybe rename this test (as it is using fqdn and not serial number to get the device info).   

7.3 `test_cv_device_tools.py`: Other issues found not linked to the refactoring - I will open an issue separetly. 

7.4 `test_cv_device_tools_generic.py`: OK 

7.5 `test_cv_device_tools_serial.py`: 
* FAILED system/test_cv_device_tools_serial.py::TestCvDeviceToolsWithSerial::test_device_is_present_by_serial[CV_DEVICE0] - AssertionError: assert False is True
Search mode should be set to FIELD_SERIAL: `assert self.inventory.is_device_exist(device_lookup=CV_DEVICE[FIELD_SERIAL], search_mode=FIELD_SERIAL) is True
` (line 138)

* FAILED system/test_cv_device_tools_serial.py::TestCvDeviceToolsWithSerial::test_configlet_apply[CV_DEVICE0] - NameError: name 'CHECK_MODE' is not defined 
Undefined variable

* FAILED system/test_cv_device_tools_serial.py::TestCvDeviceToolsWithSerial::test_device_move[CV_DEVICE0] - NameError: name 'CONTAINER_DESTINATION' is not defined
Undefined variable


Thanks a lot for your help with this!

**Comment by titom73** [comment]
Todo for next Pytest PR:

- [ ] Configure correct config file and secure content
- [ ] Build system test cases with test CV
- [ ] Cleanup `lib/json_data.py` and `lib/parametrize.py`
- [ ] Configure GH Actions to run pytest

File: tests/lib/config.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+"""
+    config.py - Declares the credential of the specified server
+"""
+username = "ansible"
```
Comment (titom73) [inline tests/lib/config.py]: We could leverage ENV variables for usernae / password / server so credentials would not be exposed and will be set per user easily ?

```python
import os

username = os.gentenv('ARISTA_AVD_CV_USER', '')
password = os.gentenv('ARISTA_AVD_CV_TOKEN', '')
server = os.gentenv('ARISTA_AVD_CV_SERVER', '')
```

**Review by titom73 — APPROVED** [review]
LGTM!

**Review by guillaumeVilar — APPROVED** [review]
LGTM!

**Pull request merged**